src/eradius_client.erl

```diff
@@ -129,8 +131,21 @@ send_remote_request(_Node, {_IP, _Port, _Secret}, _Request, _Options) ->
     error(badarg).
 
 proceed_response(Request, {ok, Response, Secret, Authenticator}, _Peer = {_ServerName, {ServerIP, Port}}, TS1, MetricsInfo) ->
-    update_client_request(Request#radius_request.cmd, MetricsInfo, eradius_metrics:timestamp(milli_seconds) - TS1),
+    update_client_request(Request#radius_request.cmd, MetricsInfo, eradius_lib:timestamp(milli_seconds) - TS1),
+    update_client_responses(MetricsInfo),
     case eradius_lib:decode_request(Response, Secret, Authenticator) of
+        {bad_pdu, "Message-Authenticator Attribute is invalid" = Reason} ->
```

Comment:
I'm less that trilled by this. Matching on what is supposed the human readable reason is IMHO a bad idea.
However, changing the bad_pdu return is a topic for another PR

Reply:
I agree, I just didn't know should I do it here or in separate PR. Here is human readable `reason` - to do minimal changes in `eradius_lib` - as it is mostly not related to metrics.

Reply:
I would opt for another PR

Reply:
yes, will fix this in a separate PR